src/algo/uint_tuple.h

```diff
@@ -103,6 +100,15 @@ struct uint_tuple {
   using storage_type =
       uint_t<_uint_tuple::round_to_possible_size((... + sizes))>;
 
+  friend bool operator==(uint_tuple x, uint_tuple y) {
+    return x.data == y.data;
+  }
+  friend bool operator!=(uint_tuple x, uint_tuple y) { return !(x == y); }
+  friend bool operator<(uint_tuple x, uint_tuple y) { return x.data < y.data; }
+  friend bool operator<=(uint_tuple x, uint_tuple y) { return !(y < x); }
+  friend bool operator>(uint_tuple x, uint_tuple y) { return y < x; }
+  friend bool operator>=(uint_tuple x, uint_tuple y) { return !(x < y); }
```

Comment:
O_o - comparison that accept by value.
Well - not conventional, but makes sense.